src/archive.cpp

```diff
@@ -188,7 +188,7 @@ namespace zim
     auto frontEntryCount = m_impl->getFrontEntryCount().v;
     int watchdog = 42;
     while(--watchdog) {
-      auto idx = randomNumber(frontEntryCount);
+      auto idx = randomNumber(frontEntryCount-1);
       auto entry =  getEntryByTitle(idx);
       auto item = entry.getItem(true);
```

Comment:
The (not so important) corner case of an empty ZIM archive (or, rather, a ZIM archive without any front entries) remained unhandled.